modules/instance_template/main.tf

```diff
@@ -18,52 +18,93 @@
 # Data Sources
 ###############
 data "google_compute_image" "image" {
-  project = "${var.source_image != "" ? var.source_image_project : "centos-cloud"}"
-  name    = "${var.source_image != "" ? var.source_image : "centos-6-v20180716"}"
+  project = var.source_image != "" ? var.source_image_project : "centos-cloud"
+  name    = var.source_image != "" ? var.source_image : "centos-6-v20180716"
 }
 
 data "google_compute_image" "image_family" {
-  project = "${var.source_image_family != "" ? var.source_image_project : "centos-cloud"}"
-  family  = "${var.source_image_family != "" ? var.source_image_family : "centos-6"}"
+  project = var.source_image_family != "" ? var.source_image_project : "centos-cloud"
+  family  = var.source_image_family != "" ? var.source_image_family : "centos-6"
 }
 
 #########
 # Locals
 #########
 
 locals {
-  boot_disk = [{
-    source_image = "${var.source_image != "" ? data.google_compute_image.image.self_link : data.google_compute_image.image_family.self_link}"
-    disk_size_gb = "${var.disk_size_gb}"
-    disk_type    = "${var.disk_type}"
-    auto_delete  = "${var.auto_delete}"
-    boot         = "true"
-  }]
+  boot_disk = [
+    {
+      source_image = var.source_image != "" ? data.google_compute_image.image.self_link : data.google_compute_image.image_family.self_link
+      disk_size_gb = var.disk_size_gb
+      disk_type    = var.disk_type
+      auto_delete  = var.auto_delete
+      boot         = "true"
+    },
+  ]
 
-  all_disks = "${concat(local.boot_disk, var.additional_disks)}"
+  all_disks = concat(local.boot_disk, var.additional_disks)
 }
 
 ####################
 # Instance Template
 ####################
 resource "google_compute_instance_template" "tpl" {
   name_prefix             = "${var.name_prefix}-"
-  machine_type            = "${var.machine_type}"
-  labels                  = "${var.labels}"
-  metadata                = "${var.metadata}"
-  tags                    = "${var.tags}"
-  can_ip_forward          = "${var.can_ip_forward}"
-  metadata_startup_script = "${var.startup_script}"
-  disk                    = ["${local.all_disks}"]
-  service_account         = ["${var.service_account}"]
+  machine_type            = var.machine_type
+  labels                  = var.labels
+  metadata                = var.metadata
+  tags                    = var.tags
+  can_ip_forward          = var.can_ip_forward
+  metadata_startup_script = var.startup_script
+  dynamic "disk" {
+    for_each = local.all_disks
+    content {
+      # TF-UPGRADE-TODO: The automatic upgrade tool can't predict
+      # which keys might be set in maps assigned here, so it has
+      # produced a comprehensive set here. Consider simplifying
+      # this after confirming which keys can be set in practice.
+
+      auto_delete  = lookup(disk.value, "auto_delete", null)
```

Comment:
Please check if you need all these fields

Reply:
I'll check